Faceit eloer

kerpakich opened the pull request.

kerpakich committed:
Add checking elo difference for players in the list

kerpakich committed:
Added checking match count for a player

kerpakich committed:
Fixed incorrect condition in line 32

kerpakich committed:
Removed unnecessary variable from lines 48-49

fedorkru commented on src/main.py:
```diff
@@ -1,23 +1,62 @@
-def get_user_data(user_id):
-    faceit_response = requests.get(f"https://open.faceit.com/data/v4/players/{user_id}",
+def get_user_data(user_id, params):
+    request_string = f"https://open.faceit.com/data/v4/players/{user_id}" + params
```
Желательно эту строку вынести в константу вне метода. А сам метод в отдельный файл. Например, faceit-api или faceitApi(не знаю какая схема наименований приветствуется в питоне).

fedorkru replied:
Константу в тот же файл, естественно.

fedorkru commented on src/main.py:
```diff
@@ -1,23 +1,62 @@
-    faceit_response = requests.get(f"https://open.faceit.com/data/v4/players/{user_id}",
+def get_user_data(user_id, params):
+    request_string = f"https://open.faceit.com/data/v4/players/{user_id}" + params
+    faceit_response = requests.get(request_string,
```
Это тоже в отдельный метод, так как потом будем переиспользовать для других вызовов api.

fedorkru commented on src/main.py:
```diff
@@ -1,23 +1,62 @@
+    player_updated_elo = player_updated_info["games"]["csgo"]["faceit_elo"]
+
+    if player_current_elo != player_updated_elo:
+        player_elo_difference = player_current_elo - player_updated_elo
```
Если эло поднялось, то здесь будет отрицательное значение.

kerpakich replied:
Верно. Перепутал местами переменные, поправлено.

fedorkru commented on src/main.py:
```diff
@@ -1,23 +1,62 @@
+
+    return player_elo_difference
+
+
```
Ниже просто два цикла подряд. Один заполняет пользователей текущими значениями, а другой сразу обновляет.
1. Стараемся все инкапсулировать в методы. Эти циклы не должны быть просто в файле - складываем в методы.
2. Два метода. initGamers - создает список пользователей. Вызывается при старте один раз. Второй updateGamers - вызывается с какой-то периодичностью и обновляет эло(пока только эло).
3. class Player - объявить поле eloDif, по умолчанию 0. Объявить метод setElo - автоматически обновляет и поле elo и поле eloDif. setElo - вызывается в updateGamers.

fedorkru commented on src/main.py:
```diff
@@ -1,23 +1,62 @@
+
 for player in players_ids:
-    user_info = get_user_data(player)
+    # Getting user information from faceit endpoint (no additional parameters needed)
```
И нужна общая терминология. Мне players(плеерс) чет ухо режет. Я предлагаю либо везде users, либо gamers везде. Сейчас в коде вижу users и players.

kerpakich replied:
Давайте останемся с get_player_data().

fedorkru commented on src/model/faceit_player.py:
```diff
@@ -1,8 +1,10 @@
 class Player:
-    def __init__(self, name, id, elo):
+    def __init__(self, name, id, elo, matches_count):
         self.player_name = name
```
Не думаю, что у каждого поля нужен префикс palyer_. Название класса уже об этом все говорит

kerpakich replied:
Поправил.

kerpakich committed:
1. Updated faceit_player class: removed "player" prefix for class fields.
2. Add "FACEIT_API_URL" const
3. Switched sides "player_updated_elo - player_current_elo" vars.
4. Moved Faceit API requests to an additional file "faceit_api"
5. Added env file for faceit_api_key to store in separate config file.

kerpakich committed:
Forgot to remove FACEIT_API and and requests usage from main.py - fixed

wingrabbit reviewed (approved):
Январь
Февраль
Март
Апрель
...
Пора!
